agent/lib/kontena/workers/service_pod_manager.rb

```diff
@@ -44,10 +44,7 @@ def populate_workers_from_master
       exclusive {
         request = rpc_client.request("/node_service_pods/list", [node.id])
         response = request.value
-        unless response['service_pods'].is_a?(Array)
-          warn "failed to get list of service pods from master: #{response['error']}"
-          return
-        end
+        raise "Invalid response" unless response['service_pods'].is_a?(Array)
```

Comment:
Why are we crashing here? This is not a rpc handler so no one is listening for this failure?